src/espIdf/partition-table/index.ts

```diff
@@ -0,0 +1,175 @@
+ * 
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  WebviewPanel,
+  window,
+  ViewColumn,
+  Uri,
+  Disposable,
+  Webview,
+} from "vscode";
+import { join } from "path";
+import { ESP } from "../../config";
+import { readFileSync } from "../../utils";
+import { writeFile } from "fs-extra";
+import { Logger } from "../../logger/logger";
+
+export class PartitionTableEditorPanel {
+  private static instance: PartitionTableEditorPanel;
+
+  private readonly panel: WebviewPanel;
+  private disposable: Disposable[] = [];
+  private readonly extensionPath: string;
+
+  private filePath: string;
+
+  public static show(extensionPath: string, filePath: string) {
+    const column = window.activeTextEditor
+      ? window.activeTextEditor.viewColumn
+      : undefined;
+    if (!!PartitionTableEditorPanel.instance) {
+      if (PartitionTableEditorPanel.instance.filePath === filePath) {
+        return PartitionTableEditorPanel.instance.panel.reveal(column);
+      }
+      // new filepath so update the webview
+      PartitionTableEditorPanel.instance.getCSVFrom(filePath).then((csv) => {
+        PartitionTableEditorPanel.instance.filePath = filePath;
+        PartitionTableEditorPanel.instance.initDataToWebview(csv);
+      });
+      return;
+    }
+    const panel = window.createWebviewPanel(
+      ESP.Webview.PartitionTableEditor.ViewType,
+      ESP.Webview.PartitionTableEditor.Title,
+      column || ViewColumn.One,
+      {
+        enableScripts: true,
+        retainContextWhenHidden: true,
+        localResourceRoots: [Uri.file(join(extensionPath, "dist", "views"))],
+        enableCommandUris: true,
+      }
+    );
+    PartitionTableEditorPanel.instance = new PartitionTableEditorPanel(
+      panel,
+      extensionPath,
+      filePath
+    );
+  }
+
+  public dispose() {
+    PartitionTableEditorPanel.instance = null;
+    this.panel.dispose();
+    while (this.disposable.length) {
+      const d = this.disposable.pop();
+      d ? d.dispose() : null;
+    }
+  }
+
+  private constructor(
+    panel: WebviewPanel,
+    extensionPath: string,
+    filePath: string
+  ) {
+    this.panel = panel;
+    this.extensionPath = extensionPath;
+    this.filePath = filePath;
+    this.panel.onDidDispose(() => this.dispose(), null, this.disposable);
+    this.panel.webview.onDidReceiveMessage(
+      (e) => this.onMessage(e),
+      null,
+      this.disposable
+    );
+    this.panel.iconPath = Uri.file(
+      join(extensionPath, "media", "espressif_icon.png")
+    );
+    this.panel.webview.html = this.initWebView(this.panel.webview);
+    this.getCSVFrom(this.filePath).then((csv) => {
```

Comment:
Should the initial data load be triggered by the webview when is ready ?

Reply:
Yes we have that from [here](https://github.com/espressif/vscode-esp-idf-extension/pull/170/files#diff-d150927ba39e0f7164cdf9c463db0796b4452dd4f139c530fe17793b471c026fR137)